internal/handler/wallet/sign.go

```diff
@@ -360,21 +363,28 @@ func (w *walletSign) privateKeyFromCache(
 		return nil, fmt.Errorf("%w: %w", utils.ErrCache, err)
 	}
 
-	// convert interface{} into byte[]
-	buf := new(bytes.Buffer)
+	if value == nil {
+		return nil, fmt.Errorf("%w: %s", utils.ErrCache, "value is nil")
+	}
 
-	err = binary.Write(buf, binary.LittleEndian, value)
-	if err != nil {
-		return nil, fmt.Errorf("%w: %w", utils.ErrCache, err)
+	byteValue, ok := value.([]byte)
+	if !ok {
+		return nil, fmt.Errorf("%w: %s", utils.ErrCache, "value is not a byte array")
 	}
 
-	cipheredPrivateKey := memguard.NewBufferFromBytes(buf.Bytes())
+	cacheValue := make([]byte, ed25519.PrivateKeySize)
+	copy(cacheValue, byteValue)
```

Comment:
are you sure this copy is needed ? i don't understand why

Reply:
ho, let me try without

Reply:
> are you sure this copy is needed ? i don't understand why

+1

Reply:
it's needed, I try without and the second time we retrieve the ciphered private key from the cache, the unciphered private key is not the same, so the signature is invalid.

Reply:
before we did: 

```go
buf := new(bytes.Buffer)
err = binary.Write(buf, binary.LittleEndian, value)
```

so now
```
cacheValue := make([]byte, ed25519.PrivateKeySize)
copy(cacheValue, byteValue)
```

is equivalent but better because we specify the length of the byte array and it's more readable than `binary.Write(buf, binary.LittleEndian, value)`